configuration/src/main/java/io/airlift/configuration/secrets/NoOpSecretsResolver.java

```diff
@@ -0,0 +1,25 @@
+package io.airlift.configuration.secrets;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public class NoOpSecretsResolver
+        extends SecretsResolver
+{
+    public NoOpSecretsResolver()
+    {
+        super(Map.of());
+    }
+
+    @Override
+    public Map<String, String> getResolvedConfiguration(Map<String, String> properties)
+    {
+        return properties;
```

Comment:
I don't think this is the correct behavior. If we enable ENV processing, then we will fail for any other provider. With this `NoOpSecretsResolver`, we disable interpolation entirely, including for ENV. Given that this is a backwards incompatible change, it seems better to fail for all providers.

So instead of `NoOpSecretsResolver`, we can use `new SecretsResolver(Map.of())`.